src/components/InputProblem.tsx

```diff
@@ -105,11 +121,54 @@ export const InputProblem = ({
     return trimmedInput.toLowerCase() === trimmedAnswer.toLowerCase()
   }
 
+  const setPersistedState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        return
+      }
+
+      if (response !== '' || inputDisabled || retriesAllowed !== 0) {
+        const newPersistedData: PersistorData = { userResponse: response, inputDisabled, retriesAllowed, feedback }
+        await persistor.put(contentId, JSON.stringify(newPersistedData))
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  const getPersistedState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        setPersistorGetStatus(PersistorGetStatus.Success)
+        return
+      }
+
+      const persistedState = await persistor.get(contentId)
+      if (persistedState !== null) {
+        const parsedPersistedState = JSON.parse(persistedState)
+        setResponse(parsedPersistedState.userResponse)
+        setInputDisabled(parsedPersistedState.inputDisabled)
+        setRetriesAllowed(parsedPersistedState.retriesAllowed)
+        setFeedback(parsedPersistedState.feedback)
+      }
+      setPersistorGetStatus(PersistorGetStatus.Success)
+    } catch (err) {
+      setPersistorGetStatus(PersistorGetStatus.Failure)
+    }
+  }
+
+  useEffect(() => {
+    setPersistedState().catch(() => { })
+    getPersistedState().catch(() => { })
+  }, [response, inputDisabled, retriesAllowed])
```

Comment:
This is problematic because the `useEffect()` will do a get more than once, which we should never have to do. At the very least, I think we should separate that one out to ensure there is only ever one `.get()` call invoked by a component. It's also a bit odd because we do a `.put()` before we do a `.get()` (it only happens to work out I think because of the conditional `response !== ''` :man_shrugging: )

The use of a `.put()` seems to be a different pattern than `UserInputBlock` and has unfortunate consequences per [your comment](https://github.com/openstax/k12-apps-raise/pull/115#issuecomment-1731754344). Perhaps this was motivated to handle the clearing of feedback in `clearFeeback()` (maybe?), but I don't think that's necessary. See related comment in `handleSubmit()`.